command/agent/auth/token-file/token_file.go

```diff
@@ -0,0 +1,101 @@
+package token_file
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-secure-stdlib/parseutil"
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/command/agent/auth"
+)
+
+type tokenFileMethod struct {
+	logger    hclog.Logger
+	mountPath string
+
+	cachedToken                 string
+	tokenFilePath               string
+	removeTokenFileAfterReading bool
```

Comment:
I raised this on the docs PR too.  I don't think we should offer this feature: for one, there's no valid dev case for it that I can think of, and we're presently positioning this as a dev feature.  Moreover, it's always risky to delete files.  I try to avoid writing code that deletes files whenever I can, unless it was a file my code created itself.  Whether an attacker or user error is involved, it sucks to be complicit in undesired data loss.

Reply:
I'm inclined to leave it in for symmetry with our other auth methods - as all of our other file-based auto-auth methods offer this. It would feel a bit inelegant/incomplete if one was without, in my view. There's also likely a good chance this is asked for later, and it would be a shame to have to reimplement code we had and decided to delete. 

All this being said, this is just my inclination, and while I feel like the method is overall better with this option, I definitely see your point of view, and I'm happy to remove it if you feel strongly about it.

Reply:
I don't think we should be seeking symmetry here.  Unless you've found a legit prod use case, and I'm incorrect that we want to discourage using this in prod?

Reply:
I definitely think we want to discourage using this in production, and we note as much, but this option feels like it adds to the offering regardless. There's a good reason we offer this in our other methods, as it's seen as more secure for some use-cases.

To put it another way: I don't think removing this option (or any amount of documentation) will stop some user somewhere trying to use it in a way they probably shouldn't be. If someone is going to do that thing anyway, I'd rather let them do it in the most secure way possible.

So, rephrasing, it's less that I want to include this option to encourage customers to do something, it's just that I think someone will probably try it anyway, and if they do, I'd rather they do things securely and properly when they do (instead of hacking a token file delete into a template `exec` or something, which would be similar, but worse). This might be the wrong way to look at things, but it's what drew me to implementing this option despite my belief that this shouldn't be the preferred way to do things in prod.

Reply:
But that's the thing, the reason we don't want them to use it in prod is that it's inherently insecure compared to the alternatives.  Let's not kid ourselves or them about that, let's not enable someone to fool themselves into thinking this is good enough.  Yes, people will still do that despite our best efforts, but by providing this option, you're making us look conflicted or inconsistent in our position.

We can always change our minds later and add it.  If we decide we want to take the approach of providing mitigations to make this less unsafe to use in prod, that can be another effort, and we can also support response wrapped tokens, which would be a big improvement for secure token distribution.  But it's so much harder to remove something after the fact than it is to add it.

Reply:
I'm happy to remove it if you'd prefer - like I've said, I do definitely see your point of view, and I could go either way on this one. I do lean towards keeping, but I'm happy to remove it!

Reply:
Yes please remove it.

Reply:
Thanks for the input, and I do appreciate the insight. Removed!